lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/file/download/BulkDownloadDAOImplTest.java

```diff
@@ -0,0 +1,312 @@
+		FileHandleAssociation fha = new FileHandleAssociation();
+		fha.setAssociateObjectId("123");
+		fha.setAssociateObjectType(null);
+		fha.setFileHandleId("456");
+		// call under test
+		BulkDownloadDAOImpl.createDBO(userId, fha);
+	}
+	
+	@Test (expected=IllegalArgumentException.class)
+	public void testCreateDBOFileHandleAssociationFileNull() {
+		FileHandleAssociation fha = new FileHandleAssociation();
+		fha.setAssociateObjectId("123");
+		fha.setAssociateObjectType(FileHandleAssociateType.FileEntity);
+		fha.setFileHandleId(null);
+		// call under test
+		BulkDownloadDAOImpl.createDBO(userId, fha);
+	}
+
+	@Test
+	public void testCreateDBOFileHandleAssociationList() {
+		FileHandleAssociation fha = new FileHandleAssociation();
+		fha.setAssociateObjectId("123");
+		fha.setAssociateObjectType(FileHandleAssociateType.FileEntity);
+		fha.setFileHandleId("456");
+		List<FileHandleAssociation> list = Lists.newArrayList(fha);
+		// call under test
+		List<DBODownloadListItem> items = BulkDownloadDAOImpl.createDBO(userId, list);
+		assertNotNull(items);
+		assertEquals(1, items.size());
+		DBODownloadListItem item = items.get(0);
+		assertEquals(new Long(123), item.getAssociatedObjectId());
+		assertEquals(FileHandleAssociateType.FileEntity.name(), item.getAssociatedObjectType());
+		assertEquals(new Long(456), item.getFileHandleId());
+		assertEquals(userId, item.getPrincipalId());
+	}
+	
+	@Test (expected=IllegalArgumentException.class)
+	public void testCreateDBOFileHandleAssociationListNullList() {
+		List<FileHandleAssociation> list = null;
+		// call under test
+		BulkDownloadDAOImpl.createDBO(userId, list);
+	}
+	
+	@Test (expected=IllegalArgumentException.class)
+	public void testCreateDBOFileHandleAssociationListEmpty() {
+		List<FileHandleAssociation> list = new LinkedList<>();
+		// call under test
+		BulkDownloadDAOImpl.createDBO(userId, list);
+	}
+	
+	@Test
+	public void testTouchUsersDownloadList() throws InterruptedException {
+		DownloadList old = bulkDownlaodDao.getUsersDownloadList(userIdString);
+		Thread.sleep(10L);
+		assertNotNull(old);
+		// call under test
+		bulkDownlaodDao.touchUsersDownloadList(userId);
+		
+		DownloadList updated = bulkDownlaodDao.getUsersDownloadList(userIdString);
+		assertNotNull(updated.getEtag());
+		assertNotNull(updated.getUpdatedOn());
+		assertFalse(updated.getEtag().equals(old.getEtag()));
+		assertTrue(updated.getUpdatedOn().getTime() > old.getUpdatedOn().getTime());
+	}
+	
+	@Test
+	public void testAddFiles() {
+		// Add a few files to a user's list
+		List<FileHandleAssociation> toAdd = fileHandleAssociations.subList(1, 3);
+		// call under test
+		DownloadList result = bulkDownlaodDao.addFilesToDownloadList(userIdString, toAdd);
+		assertNotNull(result);
+		assertEquals(userIdString, result.getOwnerId());
+		assertNotNull(result.getEtag());
+		assertNotNull(result.getUpdatedOn());
+		assertEquals(toAdd, result.getFilesToDownload());
+	}
```

Comment:
add test add the same duplicate file to the same user.
add test same file to multiple users.